qt-models/filtermodels.cpp

```diff
@@ -16,57 +16,48 @@
 #include <algorithm>
 
 namespace {
-	bool hasTag(const QStringList tags, const struct dive *d)
+	// Check if a string-list contains at least one string containing the second argument.
+	// Comparison is non case sensitive and removes white space.
+	bool listContainsSuperstring(const QStringList &list, const QString &s)
 	{
-		if (!tags.isEmpty()) {
-			auto dive_tags = get_taglist_string(d->tag_list).split(",");
-			bool found_tag = false;
-			for (const auto& filter_tag : tags)
-				for (const auto& dive_tag : dive_tags)
-					if (dive_tag.trimmed().toUpper().contains(filter_tag.trimmed().toUpper()))
-						found_tag = true;
-
-			return found_tag;
-		}
-		return true;
+		return std::any_of(list.begin(), list.end(), [&s](const QString &s2)
+				   { return s2.trimmed().contains(s.trimmed(), Qt::CaseInsensitive); });
```

Comment:
ok, this is a use of lambda I approve of. Even I can read and understand this one. :-)